frame/ethereum/src/lib.rs

```diff
@@ -738,6 +771,29 @@ impl<T: Config> Pallet<T> {
 
 		let is_transactional = true;
 		let validate = false;
+
+		let (transaction_len, weight_limit) =
+			match <T as pallet_evm::Config>::GasWeightMapping::gas_to_weight(
+				gas_limit.low_u64(),
```

Comment:
Should this use a checked conversion?

Reply:
do you mean a saturating conversion? or reject any transaction with a gas limit > u64::max?

Reply:
I guess saturating would be fine, it *shouldn't* happen so this would be a defensive thing :)